# Import des librairies et du dataset

In [625]:
import pandas as pd
pd.options.display.max_columns=None
import plotly.express as px
import numpy as np
from sklearn.preprocessing import StandardScaler
from scipy import stats
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest

In [626]:
data = pd.read_csv('data/Speed_Dating_Data.csv', encoding='cp1252')
data.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

In [627]:
data.describe()

iid           id       gender          idg       condtn  \
count  8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
mean    283.675937     8.960248     0.500597    17.327166     1.828837   
std     158.583367     5.491329     0.500029    10.940735     0.376673   
min       1.000000     1.000000     0.000000     1.000000     1.000000   
25%     154.000000     4.000000     0.000000     8.000000     2.000000   
50%     281.000000     8.000000     1.000000    16.000000     2.000000   
75%     407.000000    13.000000     1.000000    26.000000     2.000000   
max     552.000000    22.000000     1.000000    44.000000     2.000000   

              wave        round     position     positin1        order  \
count  8378.000000  8378.000000  8378.000000  6532.000000  8378.000000   
mean     11.350919    16.872046     9.042731     9.295775     8.927668   
std       5.995903     4.358458     5.514939     5.650199     5.477009   
min       1.000000     5.000000     1.000000     1.000000     1.000000   
25%       7.000000    14.000000     4.000000     4.000000     4.000000   
50%      11.000000    18.000000     8.000000     9.000000     8.000000   
75%      15.000000    20.000000    13.000000    14.000000    13.000000   
max      21.000000    22.000000    22.000000    22.000000    22.000000   

           partner          pid        match     int_corr     samerace  \
count  8378.000000  8368.000000  8378.000000  8220.000000  8378.000000   
mean      8.963595   283.863767     0.164717     0.196010     0.395799   
std       5.491068   158.584899     0.370947     0.303539     0.489051   
min       1.000000     1.000000     0.000000    -0.830000     0.000000   
25%       4.000000   154.000000     0.000000    -0.020000     0.000000   
50%       8.000000   281.000000     0.000000     0.210000     0.000000   
75%      13.000000   408.000000     0.000000     0.430000     1.000000   
max      22.000000   552.000000     1.000000     0.910000     1.000000   

             age_o       race_o     pf_o_att     pf_o_sin     pf_o_int  \
count  8274.000000  8305.000000  8289.000000  8289.000000  8289.000000   
mean     26.364999     2.756653    22.495347    17.396867    20.270759   
std       3.563648     1.230689    12.569802     7.044003     6.782895   
min      18.000000     1.000000     0.000000     0.000000     0.000000   
25%      24.000000     2.000000    15.000000    15.000000    17.390000   
50%      26.000000     2.000000    20.000000    18.370000    20.000000   
75%      28.000000     4.000000    25.000000    20.000000    23.810000   
max      55.000000     6.000000   100.000000    60.000000    50.000000   

          pf_o_fun     pf_o_amb     pf_o_sha        dec_o       attr_o  \
count  8280.000000  8271.000000  8249.000000  8378.000000  8166.000000   
mean     17.459714    10.685375    11.845930     0.419551     6.190411   
std       6.085526     6.126544     6.362746     0.493515     1.950305   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%      15.000000     5.000000     9.520000     0.000000     5.000000   
50%      18.000000    10.000000    10.640000     0.000000     6.000000   
75%      20.000000    15.000000    16.000000     1.000000     8.000000   
max      50.000000    53.000000    30.000000     1.000000    10.500000   

            sinc_o      intel_o        fun_o        amb_o       shar_o  \
count  8091.000000  8072.000000  8018.000000  7656.000000  7302.000000   
mean      7.175256     7.369301     6.400599     6.778409     5.474870   
std       1.740575     1.550501     1.954078     1.794080     2.156163   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       6.000000     6.000000     5.000000     6.000000     4.000000   
50%       7.000000     7.000000     7.000000     7.000000     6.000000   
75%       8.000000     8.000000     8.000000     8.000000     7.000000   
max      10.000000    10.000000    11.000000    10.000000    10.000000   

            l

Notes des informations extraites de la documentation:

- "attr" correspond à ce que le sondé a pensé de son partenaire.
- "attr1_1" correspond à ce que cherche le sondé.
- "attr4_1" correspond à ce que le sondé pense que la plupart des personnes du même sexe que lui cherchent chez le sexe opposé.
- "attr2_1" correspond à ce que le sondé pense que le sexe opposé recherche.
- "attr3_1" correspond à ce que le sondé pense de lui-même.
- "attr5_1" correspond à ce que le sondé pense que l'on pense de lui.
- "match" : 1 = oui, 0 = non.
- "dec" : 1 = veut revoir la personne, 0 = ne veut pas revoir la personne.
- "dec_o" : 1 = son partenaire veut le revoir, 0 = son partenaire ne veut pas le revoir.

# Recherche des attributs désirables des partenaires masculins

In [628]:
#Trier pour ne garder que les femmes
data_women = data[data['gender']==0]
data_women.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

In [629]:
#Sélectionner les colonnes correspondant à ce que les femmes affirment rechercher chez les hommes
colonnes = ['attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1']

#Calcul des notes moyennes
moyennes = data_women[colonnes].mean().reset_index().sort_values(by=0, ascending=False)
moyennes.columns = ['colonnes', 'valeurs_moyennes']
moyennes

colonnes  valeurs_moyennes
2  intel1_1         21.002502
1   sinc1_1         18.305008
0   attr1_1         18.055224
3    fun1_1         17.147292
4    amb1_1         12.827222
5   shar1_1         12.704194

In [630]:
#Graphique
fig = px.bar(
    moyennes,
    x='colonnes',
    y='valeurs_moyennes',
    title='Valeurs moyennes des qualités recherchées par les femmes (avant expérience)',
    labels={'colonnes': 'Qualités', 'valeurs_moyennes': 'Valeurs moyennes attribuées'}    
)
fig.show()

Note:

Les femmes semblent rechercher en priorité l'intelligence chez les hommes.
Il faut toutefois se méfier: il s'agit de ce que les femmes pensent chercher mais elles peuvent elles-mêmes se tromper.

In [631]:
#Couper le dataset en deux: les femmes souhaitant un second rdv et celles n'en voulant pas
data_women_dec1 = data_women[data_women['dec']==1]
data_women_dec0 = data_women[data_women['dec']==0]
data_women_dec1.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

In [632]:
#Colonnes correspondant aux notes données par les femmes à leur partenaire après rdv
colonnes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']

#Calcul de notes moyennes
moyennes_dec1 = data_women_dec1[colonnes].mean().reset_index().sort_values(by=0, ascending=False)
moyennes_dec1.columns = ['colonnes', 'valeurs_moyennes']
moyennes_dec1

colonnes  valeurs_moyennes
2    intel          7.932164
1     sinc          7.632566
3      fun          7.381710
4      amb          7.374647
0     attr          7.072787
5     shar          6.583885

In [633]:
#Graphique
fig = px.bar(
    moyennes_dec1,
    x='colonnes',
    y='valeurs_moyennes',
    title="Valeurs moyennes des notes données par les femmes à un partenaire qu'elles veulent revoir",
    labels={'colonnes': 'Qualités', 'valeurs_moyennes': 'Valeurs moyennes attribuées'}    
)
fig.show()

Note:
On voit ici que les notes sont plus étalées: c'est notamment dû à une notation différente*.
Mais l'on voit l'intelligence encore en tête et les intérêts communs en dernier. On pourrait penser que le premier est donc le facteur le plus important dans le fait qu'une femme accepte un second rendez-vous, et le second, le facteur le moins important.

Avant de conclure, observons les notes données par les femmes qui refusent un second rendez-vous.


*La notation du premier graphique se fait en répartissant 100 points entre les difgérents attributs. La notation de ce second graphe est faite en donnant une note sur 10 à chaqua attribut.

In [634]:
#Colonnes correspondant aux notes données par les femmes à leur partenaire après rdv
colonnes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']

#Calcul des moyennes
moyennes_dec0 = data_women_dec0[colonnes].mean().reset_index().sort_values(by=0, ascending=False)
moyennes_dec0.columns = ['colonnes', 'valeurs_moyennes']
moyennes_dec0

colonnes  valeurs_moyennes
2    intel          7.157363
1     sinc          6.780189
4      amb          6.704489
3      fun          5.613764
0     attr          5.231626
5     shar          4.695865

In [635]:
#Graphique
fig = px.bar(
    moyennes_dec0,
    x='colonnes',
    y='valeurs_moyennes',
    title="Valeurs moyennes des notes données par les femmes à un partenaire qu'elles ne veulent pas revoir",
    labels={'colonnes': 'Qualités', 'valeurs_moyennes': 'Valeurs moyennes attribuées'}    
)
fig.show()

Note:
On observe quelque chose de très intéressant: l'ordre des notes données aux hommes acceptés et rejetés est quasiment le même. Autrement dit, l'intelligence et la sincérité semblaient être les deux facteurs principaux pour qu'un homme soit accepté, mais l'on observe que les hommes rejetés ne sont pas mauvais dans ces deux attributs.

Creusons un peu plus.

In [636]:
#Sortir les moyennes des notes "accepté/refusé" dans le même ordre
moyennes_dec0 = moyennes_dec0.sort_values(by='colonnes')
moyennes_dec1 = moyennes_dec1.sort_values(by='colonnes')

In [637]:
#Calcul de la différence moyenne des attributs "accepté - refusé"
moyennes_dec01 = pd.DataFrame(index=moyennes_dec0.index)
moyennes_dec01['colonnes'] = moyennes_dec1['colonnes']
moyennes_dec01['valeurs'] = moyennes_dec1['valeurs_moyennes'] - moyennes_dec0['valeurs_moyennes']
moyennes_dec01 = moyennes_dec01.sort_values(by='valeurs', ascending=False)

In [638]:
#Graphique
fig = px.bar(
    moyennes_dec01,
    x='colonnes',
    y='valeurs',
    title="Différence entre les valeurs moyennes des attributs des hommes acceptés et rejetés",
    labels={'colonnes': 'Qualités', 'valeurs': 'Différence des valeurs moyennes'}    
)
fig.show()

Note:
On observe que ce qui différencie un homme accepté d'un rejeté, ce ne sont pas principalement l'intelligence et la sincérité, mais les intérêts communs, l'attractivité et le fun.

## Conclusion
Malgré des notes plus importantes pour l'intelligence et la sincérité dans la déclaration de ce que cherchent les femmes, on peut observer que les notes qu'elles donnent sur ces attributs sont relativement similaires qu'elles acceptent ou refusent un second rendez-vous.

Au contraire, les premières informations laissaient penser que les intérêts communs ne possédaient que peu d'impact dans la prise de décision, et pourtant il s'agit de l'un des trois facteurs principaux poussant une femme à accepter un second rendez-vous.

Note:
On pourra remarquer que les trois attributs impactant le moins une décision sont plus difficiles à cerner en un rendez-vous. L'intelligence, la sincérité et l'ambition se découvrent petit à petit. Pour les trois principaux facteurs, c'est le contraire. L'attractivité: on voit et donc note très rapidement la personne. Les intérêts communs: on peut découvrir assez vite au cours d'une conversation si l'on aime les mêmes choses. Le fun: il suffit de quelques secondes pour que la personne fasse rire ou non.

# Recherches des attributs désirables des partenaires féminins

In [639]:
#Trier pour ne garder que les hommes
data_men = data[data['gender']==1]

#Filtre sur les notes des attributs que les hommes disent chercher
colonnes = ['attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1']

#Calcul des moyennes
moyennes = data_men[colonnes].mean().reset_index().sort_values(by=0, ascending=False)
moyennes.columns = ['colonnes', 'valeurs_moyennes']

#Graphique
fig = px.bar(
    moyennes,
    x='colonnes',
    y='valeurs_moyennes',
    title='Valeurs moyennes des qualités recherchées par les hommes (avant expérience)',
    labels={'colonnes': 'Qualités', 'valeurs_moyennes': 'Valeurs moyennes attribuées'}    
)
fig.show()

Note:
Comme avec les femmes, on voit ici une plus grande disparité due à la différence de notation. On peut toutefois notée qu'ici ce n'est pas l'intelligence qui est en tête mais l'attractivité.

In [640]:
#Séparer le dataset en second rendez-vous acceptés vs refusés
data_men_dec1 = data_men[data_men['dec']==1]
data_men_dec0 = data_men[data_men['dec']==0]

#Notes des attributs données par les hommes à leur partenaire
colonnes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']

#Calcul des moyennes
moyennes_dec1 = data_men_dec1[colonnes].mean().reset_index().sort_values(by=0, ascending=False)
moyennes_dec1.columns = ['colonnes', 'valeurs_moyennes']
moyennes_dec1

#Graphique
fig = px.bar(
    moyennes_dec1,
    x='colonnes',
    y='valeurs_moyennes',
    title="Valeurs moyennes des qualités données par les hommes à une partenaire qu'ils veulent revoir",
    labels={'colonnes': 'Qualités', 'valeurs_moyennes': 'Valeurs moyennes attribuées'}    
)
fig.show()

Note:
L'ordre a un peu changé, mais les 4 premiers attributs sont vraiment très proches.

In [641]:
#Calcul des moyennes
moyennes_dec0 = data_men_dec0[colonnes].mean().reset_index().sort_values(by=0, ascending=False)
moyennes_dec0.columns = ['colonnes', 'valeurs_moyennes']
moyennes_dec0

colonnes  valeurs_moyennes
2    intel          6.977426
1     sinc          6.952563
4      amb          6.237808
3      fun          5.798275
0     attr          5.531739
5     shar          4.747230

In [642]:
#Graphique
fig = px.bar(
    moyennes_dec0,
    x='colonnes',
    y='valeurs_moyennes',
    title="Valeurs moyennes des qualités données par les hommes à une partenaire qu'ils ne veulent pas revoir",
    labels={'colonnes': 'Qualités', 'valeurs_moyennes': 'Valeurs moyennes attribuées'}    
)
fig.show()

Note:
Encore une fois, on peut voir que les intérêts communs sont très mal placés, à chaque fois. Et pourtant, ils pourraient là aussi être un facteur décisif.

In [643]:
#Sortir les moyennes dans le même ordre
moyennes_dec0 = moyennes_dec0.sort_values(by='colonnes')
moyennes_dec1 = moyennes_dec1.sort_values(by='colonnes')

#Calcul de la différence moyenne "acceptés - refusés"
moyennes_dec01 = pd.DataFrame(index=moyennes_dec0.index)
moyennes_dec01['colonnes'] = moyennes_dec1['colonnes']
moyennes_dec01['valeurs'] = moyennes_dec1['valeurs_moyennes'] - moyennes_dec0['valeurs_moyennes']
moyennes_dec01 = moyennes_dec01.sort_values(by='valeurs', ascending=False)

In [644]:
#Graphique
fig = px.bar(
    moyennes_dec01,
    x='colonnes',
    y='valeurs',
    title="Différence entre les valeurs moyennes des attributs des femmes acceptées et rejetées",
    labels={'colonnes': 'Qualités', 'valeurs': 'Différence des valeurs moyennes'}    
)
fig.show()

Note:
On observe que les trois attribus différenciant le plus un accord d'un refus sont les mêmes chez les hommes ou les femmes. On notera toutefois que les hommes donnent une importance toute particulière à l'attractivité.

## Conclusion :
En suivant la même logique que pour les critères féminins, on comprend que l'attractivité, les intérêts communs et le fun sont les trois principaux facteurs d'une décision positive chez un homme.

Par contre, là où les femmes accordent un intérêts relativement égale entre ces attributs, les hommes donnent une nette importance à l'attractivité.

# Attractivité : impact supposé vs impact réel

## Impact supposé

In [645]:
#Calcul des moyennes de ce que les individus pensent que les membres du même sexe cherchent.
colonnes_4_1 = ['attr4_1', 'sinc4_1', 'intel4_1', 'fun4_1', 'amb4_1', 'shar4_1']
moyennes_4_1 = data[colonnes_4_1].mean().reset_index()
moyennes_4_1.columns = ['colonnes4', 'moyennes4']
moyennes_4_1['colonnes4'] = moyennes_4_1['colonnes4'].str.replace('4_1', '')
moyennes_4_1

colonnes4  moyennes4
0      attr  26.394360
1      sinc  11.071506
2     intel  12.636308
3       fun  15.566805
4       amb   9.780089
5      shar  11.014845

In [646]:
#Calcul des moyennes de ce que les individus pensent que les membres du sexe opposé cherchent.
colonnes_2_1 = ['attr2_1', 'sinc2_1', 'intel2_1', 'fun2_1', 'amb2_1', 'shar2_1']
moyennes_2_1 = data[colonnes_2_1].mean().reset_index()
moyennes_2_1.columns = ['colonnes2', 'moyennes2']
moyennes_2_1['colonnes2'] = moyennes_2_1['colonnes2'].str.replace('2_1', '')
moyennes_2_1

colonnes2  moyennes2
0      attr  30.362192
1      sinc  13.273691
2     intel  14.416891
3       fun  18.422620
4       amb  11.744499
5      shar  11.854817

In [647]:
#Fusionner les deux puis calculer leurs moyennes
moyennes_fusion = pd.merge(moyennes_4_1, moyennes_2_1, left_on='colonnes4', right_on='colonnes2') #fusion
moyennes_fusion = moyennes_fusion.drop(columns='colonnes2')
moyennes_fusion['moyenne'] = moyennes_fusion[['moyennes4', 'moyennes2']].mean(axis=1)
moyennes_fusion = moyennes_fusion.drop(columns=['moyennes4', 'moyennes2'])
moyennes_fusion = moyennes_fusion.sort_values(by='moyenne', ascending=False)
moyennes_fusion

colonnes4    moyenne
0      attr  28.378276
3       fun  16.994712
2     intel  13.526599
1      sinc  12.172598
5      shar  11.434831
4       amb  10.762294

In [648]:
#Graphique
fig = px.bar(
    moyennes_fusion,
    x='colonnes4',
    y='moyenne',
    title='Moyenne de ce que les individus pensent que les autres individus cherchent'
)

fig.show()

Note:
On voit que les individus pensent que l'attractivité est le facteur le plus important.
Avec une note plus de 1.5 fois plus élevé que le fun et plus de 2 fois plus élevé que les autres attributs.
La différence est très nette.

## Impact réel

In [649]:
#Découper en fonction de l'accord/du refus
data_dec1 = data[data['dec']==1]
data_dec0 = data[data['dec']==0]

#Se concentrer sur les notes données suite au speed-date
colonnes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']

#Calcul des moyennes
moyennes_dec1 = data_dec1[colonnes].mean().reset_index()
moyennes_dec0 = data_dec0[colonnes].mean().reset_index()
moyennes_dec1.columns = ['colonnes', 'moyennes']
moyennes_dec0.columns = ['colonnes', 'moyennes']

#Calcul de la différence des moyennes des attributs "accord - rejet"
diff_dec = pd.DataFrame(index=moyennes_dec0.index)
diff_dec['colonnes'] = moyennes_dec1['colonnes']
diff_dec['diff'] = moyennes_dec1['moyennes'] - moyennes_dec0['moyennes']
diff_dec = diff_dec.sort_values(by='diff', ascending=False)
diff_dec

colonnes      diff
0     attr  1.917942
5     shar  1.743387
3      fun  1.635715
1     sinc  0.738225
2    intel  0.680557
4      amb  0.666144

In [650]:
#Graphique
fig = px.bar(
    diff_dec,
    x='colonnes',
    y='diff',
    title='Différence en les attributs donnés aux individus acceptés et rejetés'
)

fig.show()

Note:
On voit que l'attractivité est en moyenne ce qui différencie le plus un accord d'un rejet (tout sexe confondu).

## Conclusion:
Les gens pensent que l'attractivité est très largement l'attribut principal recherché par les autres.
Dans les faits, c'est effectivement l'attribut qui a le plus d'impact dans la prise de décision de poursuivre la relation.
L'écart entre l'impact de l'attractivité et l'impact du fun (ou des intérêts communs) semble cependant bien plus mesuré dans la réalité que dans les croyances. Mais cela peut simplement être dû à la différence de notation entre les deux mesures.

On peut donc conclure que l'attractivité, comme le pensent les gens, est l'attribut principal dans la prise de décision, mais qu'en l'état on ne peut affirmer que les gens surévaluent l'importance de l'attractivité.

# Shared interests vs shared racial background

In [651]:
#Ne garder que les colonnes sur l'origine, la décision et les intérêts communs
data_light = data.loc[:, ['samerace', 'dec', 'shar']]
data_light.describe()

samerace          dec         shar
count  8378.000000  8378.000000  7311.000000
mean      0.395799     0.419909     5.474559
std       0.489051     0.493573     2.156363
min       0.000000     0.000000     0.000000
25%       0.000000     0.000000     4.000000
50%       0.000000     0.000000     6.000000
75%       1.000000     1.000000     7.000000
max       1.000000     1.000000    10.000000

In [652]:
#Supprimer les lignes ne possédant pas de valeurs pour "shar"
data_light = data_light.dropna(subset=["shar"])
data_light.describe()

samerace          dec         shar
count  7311.000000  7311.000000  7311.000000
mean      0.394884     0.433183     5.474559
std       0.488859     0.495549     2.156363
min       0.000000     0.000000     0.000000
25%       0.000000     0.000000     4.000000
50%       0.000000     0.000000     6.000000
75%       1.000000     1.000000     7.000000
max       1.000000     1.000000    10.000000

Note:
On observe que la décision est 43.3% un accord et donc 56.7% un refus. Une baseline de prédiction serait donc de 56.7%.

In [653]:
#Séparer les variables de la target
X = data_light[['samerace', 'shar']].copy()
y = data_light['dec']

#Standardiser les données
scaler = StandardScaler()
X["shar"] = scaler.fit_transform(X[["shar"]])

#Ajouter l'intercept
X = sm.add_constant(X)

#Fit le modèle
model = sm.Logit(y, X).fit()

#Afficher les résultats
print(model.summary())

#Calculer l'accuracy
y_pred = (model.predict(X) > 0.5).astype(int)
print(f"Accuracy: {(y_pred == y).mean():.1%} | Baseline: {(1-y.mean()):.1%}")

Optimization terminated successfully.
         Current function value: 0.596686
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                    dec   No. Observations:                 7311
Model:                          Logit   Df Residuals:                     7308
Method:                           MLE   Df Model:                            2
Date:                Thu, 16 Oct 2025   Pseudo R-squ.:                  0.1279
Time:                        17:09:41   Log-Likelihood:                -4362.4
converged:                       True   LL-Null:                       -5002.1
Covariance Type:            nonrobust   LLR p-value:                1.449e-278
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.3397      0.033    -10.170      0.000      -0.405      -0.274
samerace       0.0165      0.

Note:
On a standardisé la variable 'shar', nous pouvons donc maintenant la comparer avec la variable samerace. De plus le modèle a une accuracy de 67.9%, ce qui est bien meilleur que la baseline de 56.7%, ce qui montre que les features permettent réellement de prédire la décision.

Les coefficients:
- Le coefficient de samerace est de 0.0165, donc avoir les mêmes origines n'impacte quasiment pas l'accord d'un second rendez-vous.
- Le coefficient de shar est de 0.9627, c'est environ 58 fois plus important que samerace.

Les p-values:
- La p-value de samerace est de 0.754. Donc, si samerace n'avait aucun effet réel, il y aurait 75.4% de chances d'observer un coefficient au moins aussi grand que 0.0165, juste par hasard. L'effet de samerace n'est pas statistiquement significatif.
- La p-value de shar est de 0. Il y a moins de 0.001% de chances d'obtenir un coefficient de 0.9627 si shar n'avait aucun effet. L'effet de shar est très significatif.

Les intervalles de confiance:
- L'intervalle de confiance de samerace est [-0.087, 0.120]. On ne peut pas dire si l'effet de samerace est positif, négatif ou même nul. En tout cas, avec 95% de certitude, il a un impact très modéré.
- L'intervalle de confiance de shar est de [0.903, 1.022]. Même dans le pire des cas, le coefficient est très élevé. Donc on est sûr à 95% que l'effet d'une note élevée en intérêts communs a un effet positif et fort sur l'acceptation d'un second rendez-vous.

## Conclusion
Les origines ont un impact si petit sur la prise de décision qu'on ne peut même pas affirmer qu'il existe. Donc malgré l'incertitude sur leur impact (positif/négatif/nul), les origines ont bien moins d'effet que les intérêts communs qui, eux, ont un impact fort sur l'accord d'un second rendez-vous.

# Les individus sont-ils bon à prédire leur propre valeur percue sur le marché des dates ?

In [654]:
#Grouper par iid et calculer la moyenne des notes reçues
received_ratings = data.groupby('iid')[['attr_o', "sinc_o", 'amb_o', 'fun_o', "intel_o"]].mean()

print(received_ratings.head())
print("Nombre d'individus: ", len(received_ratings))

     attr_o  sinc_o     amb_o  fun_o  intel_o
iid                                          
1       6.7     7.4  8.000000    7.2      8.0
2       7.7     7.1  7.500000    7.5      7.9
3       6.5     7.1  7.111111    6.2      7.3
4       7.0     7.1  7.700000    7.5      7.7
5       5.3     7.7  7.800000    7.2      7.6
Nombre d'individus:  551


In [655]:
#Récupérer les auto-évaluations de chaque individu
self_rating = data.groupby('iid')[["attr3_1", "sinc3_1", "amb3_1", "fun3_1", "intel3_1"]].first()

print(self_rating.head())
print("Nombre d'individus: ", len(self_rating))

     attr3_1  sinc3_1  amb3_1  fun3_1  intel3_1
iid                                            
1        6.0      8.0     7.0     8.0       8.0
2        7.0      5.0     3.0    10.0       8.0
3        8.0      9.0     8.0     8.0       9.0
4        7.0      8.0     8.0     9.0       7.0
5        6.0      3.0     8.0     6.0      10.0
Nombre d'individus:  551


In [656]:
#Calcul l'écart entre ce que pense valoir une personne et sa valeur sur le marché
gaps = self_rating.copy()
gaps['attr_gap'] = self_rating["attr3_1"] - received_ratings['attr_o']
gaps['sinc_gap'] = self_rating["sinc3_1"] - received_ratings['sinc_o']
gaps['amb_gap'] = self_rating["amb3_1"] - received_ratings['amb_o']
gaps['fun_gap'] = self_rating["fun3_1"] - received_ratings['fun_o']
gaps['intel_gap'] = self_rating["intel3_1"] - received_ratings['intel_o']

#Garder seulement les écarts
gaps = gaps[['attr_gap', 'sinc_gap', 'amb_gap', 'fun_gap', 'intel_gap']]

#Calcul des moyennes
gaps_mean = gaps.mean().reset_index()
gaps_mean = gaps_mean.sort_values(0, ascending=False)

print(gaps.head())
print("\nMoyennes des écarts: ")
print(gaps_mean)

     attr_gap  sinc_gap   amb_gap  fun_gap  intel_gap
iid                                                  
1        -0.7       0.6 -1.000000      0.8        0.0
2        -0.7      -2.1 -4.500000      2.5        0.1
3         1.5       1.9  0.888889      1.8        1.7
4         0.0       0.9  0.300000      1.5       -0.7
5         0.7      -4.7  0.200000     -1.2        2.4

Moyennes des écarts: 
       index         0
3    fun_gap  1.269518
1   sinc_gap  1.067415
4  intel_gap  0.987908
0   attr_gap  0.898060
2    amb_gap  0.757052


In [657]:
#Calcul de la moyenne des écart moyens
gaps_mean[0].mean()

np.float64(0.9959906863169274)

In [658]:
#Graphique
fig = px.bar(
    gaps_mean,
    x="index",
    y=0,
    title="Auto-évaluation - note moyenne reçue (positif = surévaluation)",
    labels={"0": "Ecart moyen"}
)

fig.show()

Note:
On a un écart moyen entre ce que les individus pensent valoir et leur valeur sur le marché de 1 (toutes catégories confondues).
En observant le graphique, on voit que cet écart moyen (positif: sur-estimation) est valable pour toutes les catégories dans un intervalle compris entre 0.75 (l'ambition) et 1.26 (le fun).

On peut dire qu'en moyenne, les individus s'évaluent 1 point plus haut que ce qu'ils ne valent sur le marché.

## Conclusion
Les individus ne sont pas bons à évaluer leur propre valeur sur le marché du date.

# Différences des décisions entre le premier et le dernier rendez-vous

In [659]:
#Filtrer en ne gardant que les colonnes nous intéressant et supprimer les lignes contenant des valeurs manquantes
clean_data = data[["iid", "pid", "order", "dec_o"]].dropna()

print("Nombre de dates après nettoyage: ", len(clean_data))
print(clean_data['order'].describe())

print(f"\nNombre de dates à l'ordre 1 (premiers): {len(clean_data[clean_data['order']==1])}")
print(f"Taux d'acceptation quand on est le premier date: {clean_data[clean_data['order']==1]['dec_o'].mean():.1%}")

Nombre de dates après nettoyage:  8368
count    8368.000000
mean        8.931764
std         5.478099
min         1.000000
25%         4.000000
50%         8.000000
75%        13.000000
max        22.000000
Name: order, dtype: float64

Nombre de dates à l'ordre 1 (premiers): 550
Taux d'acceptation quand on est le premier date: 49.1%


In [660]:
#Trouver le dernier ordre de chaque individu
last_orders = clean_data.groupby('iid')['order'].max().reset_index()
last_orders.columns = ['iid', 'last_order']

#Fusionner pour identifier les derniers dates
clean_data_with_last = clean_data.merge(last_orders, on='iid')
clean_data_with_last['is_last'] = (clean_data_with_last['order'] == clean_data_with_last['last_order'])

#Filtrer sur les derniers dates
last_dates = clean_data_with_last[clean_data_with_last['is_last']]

print("Nombre de dates à l'ordre 'dernier': ", len(last_dates))
print(f"Taux d'acceptation quand on est le dernier date: {last_dates['dec_o'].mean():.1%}")

Nombre de dates à l'ordre 'dernier':  551
Taux d'acceptation quand on est le dernier date: 45.2%


Note:
On observe une différence entre le taux d'acceptation lors d'un premier et d'un dernier rendez-vous (3.9 points).

In [661]:
def get_statistics(last=0):
    
    #Extraire les premiers rendez-vous
    first_dates = clean_data_with_last[clean_data_with_last['order'] == 1]

    #Extraire les rendez-vous d'un certain ordre
    if last == 0: #On mesure en fonction de tous les derniers rendez-vous
        last = last_dates
    else: #On donne un nombre précis pour l'ordre'
        last = clean_data_with_last[clean_data_with_last['order'] == last]

    #Nombre de succès
    successes = [first_dates['dec_o'].sum(), last['dec_o'].sum()]

    #Nombre total d'observations
    n_obs = [len(first_dates), len(last)]

    #Test de proportions
    z_stat, p_value = proportions_ztest(successes, n_obs)

    print(f"Premier: {first_dates['dec_o'].mean():.1%} ({successes[0]}/{n_obs[0]})")
    print(f"Autre: {last['dec_o'].mean():.1%} ({successes[1]}/{n_obs[1]})")
    print(f"\nDifférence: {(first_dates['dec_o'].mean() - last['dec_o'].mean())*100:.1f} points")
    print(f"P-value: {p_value:.4f}")

In [662]:
print("Test confondant tous les derniers rendez-vous:\n")
get_statistics()

Test confondant tous les derniers rendez-vous:

Premier: 49.1% (270/550)
Autre: 45.2% (249/551)

Différence: 3.9 points
P-value: 0.1949


In [663]:
print("Test se concentrant sur les rendez-vous d'ordre 10:\n")
get_statistics(10)

Test se concentrant sur les rendez-vous d'ordre 10:

Premier: 49.1% (270/550)
Autre: 41.0% (194/473)

Différence: 8.1 points
P-value: 0.0097


## Conclusion
La différence observée entre un premier date et un dernier (3.9 points) n'est pas statistiquement significatives (p-value largement supérieure à 0.05). On ne peut donc exclure que cette différence soit simplement due au hasard.
Il ne semble donc ni meilleur ni pire d'être le premier ou le dernier rendez-vous de la soirée pour obtenir un second date.

On observera qu'il existe une différence statistique entre être le premier rendez-vous de quelqu'un et être son dixième (indépendamment du fait qu'il s'agisse ou non de son dernier rendez-vous). On peut donc affirmer qu'il est préférable d'être le premier rendez-vous de quelqu'un plutôt que son dixième pour obtenir un second rendez-vous.

# Conclusion générale

Plusieurs facteurs influençant l'acceptation à un second date ont été trouvés:

- Toutes les qualités ont un impact dans la décision, mais principalement les intérêts communs, l'attractivité et le fun. Et ce peu importe le genre.

- Les intérêts communs influencent bien plus la décision que les origines, ce qui rejoint le premier point sur l'importance des intérêts communs.

- Un individu aura tendance à davantage accepter un second rendez-vous lors du premier date que lors du dixième.